client/src/common/Stopwatch.js

```diff
@@ -0,0 +1,41 @@
+class Stopwatch {
+    constructor() {
+        this.callback = null
+        this.myWorker = this.createWorker( (e) => {
+            if (e.data == 'start') {
+                clearInterval(interval)
+                let d0 = (new Date()).valueOf()
+                interval = setInterval( () => { 
+                    let d = (new Date()).valueOf()
+                    let diff = d - d0
+                    let count = Math.floor(diff/1000)
+                    self.postMessage(count)
+                }, 100)
+            }
+            if (e.data == 'stop') {
+                clearInterval(interval)
+            }
+        })
+
+        this.myWorker.onmessage = (e) => {
+            console.log('received', e.data)
+            this.callback(e.data)
+        }
+    }
+
+    createWorker(fn) {
```

Comment:
Code that instantiates the Stopwatch should never call this function, so it should not be a class method. It should be lifted to a local function.